### PR TITLE
first draft of the contributing guide (close #885)

### LINKED ISSUE #885: Create a CONTRIBUTING.md file
Create a [CONTRIBUTING.md](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors) file to have clear information, steps, and describing ways people can contribute to Kubefirst.

### PR DESCRIPTION
**Pull request by fharper**
Signed-off-by: Frédéric Harper <fred@kubeshop.io>

### CODE REVIEW

**Review by pagottoo — DISMISSED** [review]

**Comment by fharper** [comment]
Please do not merge yet, @converge will check what is the issue with [Test CI failing](https://github.com/kubefirst/kubefirst/issues/903) since it failed with this PR.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+- and, of course, any code contributions to Kubefirst, or the documentations.
+
+Before making a code change, first discuss your idea via an [issue](https://github.com/kubefirst/kubefirst/issues/new/choose). Please check if a feature request or bug report does [already exist](https://github.com/kubefirst/kubefirst/issues/) before creating a new one.
+
+## Getting Started with the Code
+
+Kubefirst is created using the [Go Programming Language](https://go.dev). To set up your computer, follow [these steps](https://go.dev/doc/install).
+
+Once Go is installed, you can run Kubefirst from any branch using `go run .`. Go will automatically install the needed modules listed in the [go.mod](go.mod) file. As an example, if you want to create a [local cluster](https://docs.kubefirst.io/kubefirst/local/install.html), the command would be `go run . local`.
+b
```
Comment (johndietz) [inline CONTRIBUTING.md:25]:
```suggestion

```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# Contributing to Kubefirst
+
+Firstly, we want to thank you for investing your precious time to contribute to Kubefirst!
```
Comment (johndietz) [inline CONTRIBUTING.md:3]:
```suggestion
Firstly, we want to thank you for investing your valuable time to contribute to Kubefirst!
```
precious comes across as sarcastic to my ear

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+_⚠️ Please note that this file is a work-in-progress, so more details will be added in the future._
+
+Note we have a [code of conduct](CODE_OF_CONDUCT.md) which needs to be followed in all your interactions with the project to keep our community healthy.
+
+## Ways to Contribute
+
+At Kubefirst, we believe that every contribution are valuable, not just the code one, which means we welcome
```
Comment (johndietz) [inline CONTRIBUTING.md:11]:
```suggestion
At Kubefirst, we believe that every contribution is valuable, not just the code one, which means we welcome
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+Kubefirst is created using the [Go Programming Language](https://go.dev). To set up your computer, follow [these steps](https://go.dev/doc/install).
+
+Once Go is installed, you can run Kubefirst from any branch using `go run .`. Go will automatically install the needed modules listed in the [go.mod](go.mod) file. As an example, if you want to create a [local cluster](https://docs.kubefirst.io/kubefirst/local/install.html), the command would be `go run . local`.
+b
+Since Go is a compiled programming language, every time you use the `run` command, Go will compile the code before running it. If you want to save time, you can compile your code using `go build`, which will generate a file named `kubefirst`. You will then be able to run your compiled version with the `./kubefirst` command.
+
+## Help
+
+If you need help in your Kubefirst journey as a contributor, please join our [Slack Community](http://kubefirst.io/slack). We have the `#kubefirst-oss` channel where you can ask any questions or get help with anything contribution-related. For support as an user, please ask in the `#helping-hands` channel, or directly to @fharper (Fred in Slack), our Developer Advocate.
```
Comment (johndietz) [inline CONTRIBUTING.md:30]:
```suggestion
If you need help in your Kubefirst journey as a contributor, please join our [Slack Community](http://kubefirst.io/slack). We have the `#kubefirst-oss` channel where you can ask any questions or get help with anything contribution-related. For support as a user, please ask in the `#helping-hands` channel, or directly to @fharper (Fred in Slack), our Developer Advocate.
```

**Review by johndietz — CHANGES_REQUESTED** [review]

**Comment by converge** [comment]
tks for the heads up @fharper ! We actually just broke the unit test, Im checking with the team if anyone can fix it.

**Comment by converge** [comment]
> @converge : so can I merge this PR or you need it to test it?

sorry for the delay, we can merge.

**Review by converge — APPROVED** [review]

**Review by pagottoo — APPROVED** [review]

**Pull request merged**